cpp/ycm/CMakeLists.txt

```diff
@@ -103,16 +114,25 @@ if ( USE_CLANG_COMPLETER AND
     execute_process( COMMAND tar -xjf ${CLANG_FILENAME} )
   elseif( CLANG_FILENAME MATCHES ".+xz" )
     execute_process( COMMAND tar -xJf ${CLANG_FILENAME} )
+  elseif( CLANG_FILENAME MATCHES ".+exe" )
+    # NSIS tool is used to generate the Clang installer on Windows. Files
+    # from the installer can be extracted with 7-Zip.
+    execute_process( COMMAND 7z x ${CLANG_FILENAME} OUTPUT_QUIET )
```

Comment:
Any chance we could not have that extra dependency? I could live with another python-only dependency in `third_party` that could do the necessary work; but we'd be shipping that then and the user wouldn't have to install it themselves.